x-pack/filebeat/input/entityanalytics/provider/azuread/conf.go

```diff
@@ -34,6 +36,11 @@ func (c *conf) Validate() error {
 	if c.UpdateInterval == 0 {
 		return errors.New("update_interval must not be zero")
 	}
+	switch strings.ToLower(c.Dataset) {
+	case "", "all", "users", "devices":
+	default:
+		return errors.New("dataset must be 'all', 'users', 'devices' or empty")
+	}
```

Comment:
Additional comment: If validation is required , then can we extend the unit test [here](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/entityanalytics/provider/azuread/conf_test.go#L19)

Reply:
Not super important may be..!